scmrepo/git/backend/dulwich/__init__.py

```diff
@@ -344,8 +345,10 @@ def is_tracked(self, path: str) -> bool:
         return False
 
     def is_dirty(self, untracked_files: bool = False) -> bool:
-        staged, unstaged, untracked = self.status()
-        return bool(staged or unstaged or (untracked_files and untracked))
+        kwargs: Dict[str, Any] = (
+            {} if untracked_files else {"untracked_files": "no"}
+        )
+        return any(self.status(**kwargs))
```

Comment:
I wonder if we should optimize this further, i.e. use generator and return False if there is any changes rather than collecting everything. :)

Something like:
```python
from itertools import chain, zip_longest

with open_repo_closing(repo) as r:
        # 1. Get status of staged
        tracked_changes = get_tree_changes(r)
        # 2. Get status of unstaged
        index = r.open_index()
        normalizer = r.get_blob_normalizer()
        filter_callback = normalizer.checkin_normalize

        unstaged_changes = get_unstaged_changes(index, r.path, filter_callback)
        untracked_paths = get_untracked_paths(
            r.path,
            r.path,
            index,
            exclude_ignored=not ignored,
            untracked_files=untracked_files,
        )
        return any(chain.from_iterable(zip_longest(s, d)))
```

The only doubt that I have is that it won't reuse IgnoreManager, which I think already happens for `status`.
No strong opinion though, we could also propose this in dulwich.